yt_dlp/__init__.py

```diff
@@ -435,6 +435,21 @@ def metadataparser_actions(f):
         elif ed and proto == 'default':
             default_downloader = ed.get_basename()
 
+    if opts.color:
+        policies = {
+            'always': True,
+            'auto': None,
+            'no_color': 'no_color',
+            'never': False,
+        }
+        result = {}
+        for name, policy in opts.color.items():
+            if policy not in policies:
+                raise ValueError(f'"{policy}" is not a valid color policy')
+            result[name] = policies[policy]
+
+        opts.color = result
```

Comment:
```suggestion
    for name, policy in opts.color.items():
        if policy not in ('always', 'auto', 'no_color', 'never'):
            raise ValueError(f'"{policy}" is not a valid color policy for {name}')
```
https://github.com/yt-dlp/yt-dlp/pull/6904#discussion_r1180397402